testing/sdk_upgrade.py

```diff
@@ -201,50 +204,18 @@ def _upgrade_or_downgrade(
             package_name, service_name))
         sdk_plan.wait_for_completed_deployment(service_name, timeout_seconds)
 
-# Retry the decorated function several times in the event of exception_class. Raise the exception
-# normally if the number of retries is exceeded.
-def _retry(retries=3, exception_class=Exception):
-    def decorator(fn):
-        @functools.wraps(fn)
-        def wrapper(*args, **kwargs):
-            exception = None
-            for i in range(retries):
-                try:
-                    result = fn(*args, **kwargs)
-                except exception_class as e:
-                    exception = e
-                    continue
-
-                return result
-
-            raise exception
-        return wrapper
 
-    return decorator
-
-@_retry(exception_class=AttributeError)
+@retrying.retry(stop_max_attempt_number=5,
+                wait_fixed=1000,
+                retry_on_result=lambda result: result is None)
 def _get_pkg_version(package_name):
-    return re.search(
+    result = re.search(
         r'"version": "(\S+)"',
-        sdk_cmd.run_cli('package describe {}'.format(package_name), print_output=False)).group(1)
-
-# Default repo is the one at index=0.
-def _add_repo(repo_name, repo_url, prev_version, index, default_repo_package_name):
-    assert shakedown.add_package_repo(
-        repo_name,
-        repo_url,
-        index)
-    # Make sure the new default repo packages are available
-    _wait_for_new_default_version(prev_version, default_repo_package_name)
-
-
-def _add_last_repo(repo_name, repo_url, prev_version, default_repo_package_name):
-    assert shakedown.add_package_repo(
-        repo_name,
-        repo_url)
-    # Make sure the new default repo packages are available
-    _wait_for_new_default_version(prev_version, default_repo_package_name)
+        sdk_cmd.run_cli('package describe {}'.format(package_name), print_output=False))
+    return None if result is None else result.group(1)
 
 
-def _wait_for_new_default_version(prev_version, default_repo_package_name):
-    shakedown.wait_for(lambda: _get_pkg_version(default_repo_package_name) != prev_version, noisy=True)
+def _wait_for_new_default_version(package_name, prev_version):
```

Comment:
Is this a sufficient condition to check? Would it not make more sense to just check that the package is being served from the newly added universe URL?

Reply:
Providing useful information like that would be very un-Cosmos-like. We're resorting to the version changing as an indicator that Cosmos actually accepted the repo list change.

Cosmos' repo handling is "asynchronous", i.e. it doesn't produce any useful indication of success nor of failure.